src/cryoem_monitor/client/logger.py

```diff
@@ -0,0 +1,171 @@
+
+
+# Classes for type hinting of XML data - Values
+class ValuePar(BaseXmlModel):
+    datatype: str = attr(name="Type")
+    value: float | int
+
+
+class ParameterValue(BaseXmlModel):
+    timestamp: str = attr(name="Timestamp")
+    value: ValuePar = element(tag="Value")
+
+
+class ParameterValues(BaseXmlModel):
+    parameter_value: List[ParameterValue] = element(tag="ParameterValue")
+
+
+class ValueThresh(BaseXmlModel):
+    datatype: str = attr(name="Type")
+    value: int | float
+
+
+class Threshold(BaseXmlModel):
+    name: Literal[
+        "WarningMin",
+        "CautionMin",
+        "CriticalMin",
+        "WarningMax",
+        "CautionMax",
+        "CriticalMax",
+    ] = attr(name="Name")
+    # All optional fields are set to None by default or an empty list
+    value: Optional[ValueThresh] = element(tag="Value", default=None)
+
+
+class Limit(BaseXmlModel, tag="Limit"):
+    timestamp: str = attr(name="Timestamp")
+    thresholds: List[Threshold] = element(tag="Threshold")
+
+
+class Limits(BaseXmlModel):
+    limit: List[Limit] = element(tag="Limit")
+
+
+class ValueData(BaseXmlModel):
+    start: str = attr(name="Start")
+    end: str = attr(name="End")
+    valueid: str = attr(name="ParameterID")
+    parameter: str = attr(name="Parameter")
+    # Sometimes, pydantic cannot identify element as part of list, so initialize
+    parameter_value: List[ParameterValues] = element(
+        tag="ParameterValues", default_factory=list
+    )
+    limits: Optional[Limits] = element(tag="Limits", default_factory=list)
+
+
+class Values(BaseXmlModel):
+    start: str = attr(name="Start")
+    end: str = attr(name="End")
+    instrument: str = attr(name="Instrument")
+    value_data: List[ValueData] = element(tag="ValueData")
+
+
+# Class for type hinting of XML data - HealthMonitor (top level)
+class HealthMonitor(
+    BaseXmlModel,
+    tag="HealthMonitor",
+):
+    enumerations: Enumerations = element(tag="Enumerations")
+    instruments: Instruments = element(tag="Instruments")
+    values: Values = element(tag="Values")
+
+
+# Load and extract required values from XML file
+with open("src/cryoem_monitor/client/HealthMonitor.xml") as file:
+    xml_data = file.read()
+
+# Remove the namespace from the XML data due to this specific one being invalid
+# Normally, this is not needed
+xml_data = xml_data.replace(
+    ' xmlns="HealthMonitorExport http://schemas.fei.com/HealthMonitor/Export/2009/07"',
+    "",
+)
+
+try:
+    m = HealthMonitor.from_xml(xml_data)
+except ValidationError as exc:
+    print(exc)
+a = 2
+
+# Save all parameter names in a list
+# parameter_names = []
+# for child in root[2]:
+#     # parameter_names.append(child.attrib["Parameter"])
+#     parameter_names.append(element.attrib["Parameter"])
+
+# Post relevant data to the server
+# url = "http://localhost:8000/set"
+
+# headers = {"type": "fig_brightness", "value": 220}
+# requests.post(url, headers=headers)
```

Comment:
I'd remove all of this for code that is going into the `main` branch. I'd keep it in a separate uncommitted file or, better, put something to this effect in a test